api/iaas.go

```diff
@@ -225,13 +225,17 @@ func templateUpdate(w http.ResponseWriter, r *http.Request, token auth.Token) (e
 		return &errors.HTTP{Code: http.StatusBadRequest, Message: err.Error()}
 	}
 	templateName := r.URL.Query().Get(":template_name")
+	iaasName := r.Form.Get("IaaSName")
```

Comment:
It already does [here](https://github.com/tsuru/tsuru/blob/master/iaas/iaas.go#L115-L124) after returning `dbTpl.Update`